### PR TITLE
[ticket/16617] Add events to override message/PM BBCode status indications

### PR DESCRIPTION
**Pull request by rxu**
Replacement for #6065.

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16617">PHPBB3-16617</a>.

### CODE REVIEW

**Comment by rxu** [comment]
!set Event

**Comment by 3D-I** [comment]
Not sure about "flash" to be honest, due to https://github.com/phpbb/phpbb/pull/6056 which I think it should be adressed to 3.3.x as well.

**Comment by 3D-I** [comment]
I'd like to add that at the moment _(I know the old PR and my question is the same)_ I can't realize the correct use of these events. Someone who can enlighten me with an example, please?

**Comment by rxu** [comment]
Well, you never know what extension author could invent.
F.e. one could add some permission or special condition for using post links and so on.

File: phpBB/includes/ucp/ucp_pm_compose.php [hunk 1]
```diff
@@ -687,6 +687,27 @@ function compose_pm($id, $mode, $action, $user_folders = array())
 	$flash_status	= ($config['auth_flash_pm'] && $auth->acl_get('u_pm_flash')) ? true : false;
 	$url_status		= ($config['allow_post_links']) ? true : false;
 
+	/**
+	 * Event to override private message BBCode status indications
+	 *
+	 * @event core.ucp_pm_compose_modify_bbcode_status_indications
```
Comment (3D-I) [inline phpBB/includes/ucp/ucp_pm_compose.php:693]: I'd remove the `_indications` word because `status` says it all.

Applies to both files.

**Comment by 3D-I** [comment]
> 
> 
> Well, you never know what extension author could invent.
> F.e. one could add some permission or special condition for using post links and so on.

Makes sense, still I am missing something ATMm but's ok.

**Review by 3D-I — APPROVED** [review]

**Pull request merged**